src/kemal/middleware/filters.cr

```diff
@@ -74,7 +74,10 @@ module Kemal::Middleware
     property block : HTTP::Server::Context -> String
 
     def initialize(&block : HTTP::Server::Context -> _)
-      @block = ->(context : HTTP::Server::Context) { block.call(context).to_s }
+      @block = ->(context : HTTP::Server::Context) do 
+        block.call(context).to_s.to_json if context.request.headers["Content-Type"]? == "application/json"
```

Comment:
Actually, I don't know if I'll have time right now, and I also like [this comment](https://github.com/sdogruyol/kemal/issues/211#issuecomment-249910211) to have an explicit way to render json (though probably adding a method to Context instead of render will be easier, because there's no implicit context object for render to work)